src/main/java/com/thealgorithms/maths/Fibonacci.java

```diff
@@ -0,0 +1,41 @@
+package com.thealgorithms.maths;
+
+import java.math.BigInteger;
+
+/**
+ * This class provides methods for calculating Fibonacci numbers using BigInteger for large values of 'n'.
+ */
+public final class Fibonacci {
```

Comment:
There is another [`Fibonacci`](https://github.com/TheAlgorithms/Java/blob/1dc64b16859d95446a8bfe1c50692ce57316bc42/src/main/java/com/thealgorithms/dynamicprogramming/Fibonacci.java) class. What do you think about renaming it to `FibonacciLoop`?